mart/attack/adversary.py

```diff
@@ -97,12 +96,19 @@ def __init__(
         self.max_iters = max_iters
         self.callbacks = OrderedDict()
 
+        # Import Callback type here to avoid circular import issues.
+        from .callbacks import Callback
```

Comment:
This looks strange. But we can remove it later if we follow the change of perturber in #29

Reply:
Agreed. Something is wrong if this is necessary.